src/transformers/pipelines/text_generation.py

```diff
@@ -173,16 +173,21 @@ def _sanitize_parameters(
         forward_params = generate_kwargs
 
         postprocess_params = {}
-        if return_full_text is not None and return_type is None:
-            if return_text is not None:
-                raise ValueError("`return_text` is mutually exclusive with `return_full_text`")
-            if return_tensors is not None:
-                raise ValueError("`return_full_text` is mutually exclusive with `return_tensors`")
-            return_type = ReturnType.FULL_TEXT if return_full_text else ReturnType.NEW_TEXT
-        if return_tensors is not None and return_type is None:
-            if return_text is not None:
-                raise ValueError("`return_text` is mutually exclusive with `return_tensors`")
+        if return_type is not None and (
+            return_text is not None or return_full_text is not None or return_tensors is not None
+        ):
+            raise ValueError(
+                "`return_type` is mutually exclusive with `return_text`, `return_full_text`, and `return_tensors`"
+            )
+        if return_tensors is not None:
+            if return_text or return_full_text:
+                raise ValueError("`return_text` and `return_full_text` are mutually exclusive with `return_tensors`")
             return_type = ReturnType.TENSORS
+        elif return_full_text and return_type is None:
+            return_type = ReturnType.FULL_TEXT
+        elif (return_text or return_full_text is not None) and return_type is None:
+            # Explicitly setting return_full_text = False is caught here, and results in ReturnType.NEW_TEXT
+            return_type = ReturnType.NEW_TEXT
```

Comment:
This whole block is really hard to parse, is it possible to add white space/comments?